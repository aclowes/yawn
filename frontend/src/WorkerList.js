import React from 'react';
import {Table, Alert} from 'react-bootstrap';
import {Link} from 'react-router';

import API from "./API";

export default class WorkerList extends React.Component {
  constructor(props) {
    super(props);
    document.title = `YAWN - Workers`;
    this.state = {workers: null, error: null};
  }

  componentDidMount() {
    API.get(`/api/workers/`, (payload, error) => {
      this.setState({workers: payload, error});
    });
  }

  renderRows() {
    if (this.state.workers === null) {
      return (
        <tr>
          <td colSpan="3">Loading...</td>
        </tr>
      )
    } else {
      return this.state.workers.map((worker) => {
        return (
          <tr key={worker.id}>
            <td><Link to={`/workers/${worker.id}`}>{worker.name}</Link></td>
            <td>{worker.status}</td>
            <td>{worker.start_timestamp}</td>
            <td>{worker.last_heartbeat}</td>
          </tr>
        )
      })
    }
  }

  render() {
    return (
      <div>
        <h3>Workers</h3>
        {this.state.error && <Alert bsStyle="danger">{this.state.error}</Alert>}
        <Table striped bordered condensed hover>
          <thead>
          <tr>
            <th>Name</th>
            <th>Status</th>
            <th>Start Time</th>
            <th>Last Heartbeat</th>
          </tr>
          </thead>
          <tbody>
          {this.renderRows()}
          </tbody>
        </Table>
      </div>
    )
  }
}
