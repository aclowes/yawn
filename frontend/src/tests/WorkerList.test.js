import React from 'react';
import renderer from 'react-test-renderer';
import {browserHistory} from 'react-router'

import WorkerList from '../WorkerList';
import {mockAPI} from './mocks'

it('WorkerList loading', () => {
  mockAPI([null]);
  const component = renderer.create(<WorkerList />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('WorkerList failure', () => {
  mockAPI();
  const component = renderer.create(<WorkerList />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('WorkerList success', () => {
  mockAPI([[{
    "id": 4,
    "name": "alec-graybeard.local 36470",
    "status": "exited",
    "start_timestamp": "2017-01-30T04:13:47.966080Z",
    "last_heartbeat": "2017-01-30T04:15:10.845289Z"
  }]]);
  const component = renderer.create(<WorkerList />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
