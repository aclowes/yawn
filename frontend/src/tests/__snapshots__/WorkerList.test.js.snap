// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkerList failure 1`] = `
<div>
  <h3>
    Workers
  </h3>
  <div
    className="alert alert-danger"
    role="alert"
  >
    some error message
  </div>
  <table
    className="table table-striped table-bordered table-condensed table-hover"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Status
        </th>
        <th>
          Start Time
        </th>
        <th>
          Last Heartbeat
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          colSpan="4"
        >
          Loading...
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`WorkerList loading 1`] = `
<div>
  <h3>
    Workers
  </h3>
  <table
    className="table table-striped table-bordered table-condensed table-hover"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Status
        </th>
        <th>
          Start Time
        </th>
        <th>
          Last Heartbeat
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          colSpan="4"
        >
          Loading...
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`WorkerList success 1`] = `
<div>
  <h3>
    Workers
  </h3>
  <table
    className="table table-striped table-bordered table-condensed table-hover"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Status
        </th>
        <th>
          Start Time
        </th>
        <th>
          Last Heartbeat
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <a
            onClick={[Function]}
            style={Object {}}
          >
            alec-graybeard.local 36470
          </a>
        </td>
        <td>
          exited
        </td>
        <td>
          2017-01-30T04:13:47.966080Z
        </td>
        <td>
          2017-01-30T04:15:10.845289Z
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
